cmd/hypper/upgrade.go

```diff
@@ -0,0 +1,193 @@
+/*
+Copyright The Helm Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"github.com/Masterminds/log-go"
+	logio "github.com/Masterminds/log-go/io"
+	"github.com/rancher-sandbox/hypper/pkg/eyecandy"
+	helmCli "helm.sh/helm/v3/pkg/cli"
+
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+
+	"github.com/rancher-sandbox/hypper/pkg/action"
+	"helm.sh/helm/v3/cmd/helm/require"
+	"helm.sh/helm/v3/pkg/chart/loader"
+	"helm.sh/helm/v3/pkg/cli/output"
+	"helm.sh/helm/v3/pkg/cli/values"
+	"helm.sh/helm/v3/pkg/getter"
+	"helm.sh/helm/v3/pkg/storage/driver"
+)
+
+const upgradeDesc = `
+This command upgrades a release to a new version of a chart.
+
+The upgrade arguments must be a release and chart. The chart
+argument can be either: a chart reference('example/mariadb'), a path to a chart directory,
+a packaged chart, or a fully qualified URL. For chart references, the latest
+version will be specified unless the '--version' flag is set.
+
+To override values in a chart, use either the '--values' flag and pass in a file
+or use the '--set' flag and pass configuration from the command line, to force string
+values, use '--set-string'. In case a value is large and therefore
+you want not to use neither '--values' nor '--set', use '--set-file' to read the
+single large value from file.
+
+You can specify the '--values'/'-f' flag multiple times. The priority will be given to the
+last (right-most) file specified. For example, if both myvalues.yaml and override.yaml
+contained a key called 'Test', the value set in override.yaml would take precedence:
+
+    $ helm upgrade -f myvalues.yaml -f override.yaml redis ./redis
+
+You can specify the '--set' flag multiple times. The priority will be given to the
+last (right-most) set specified. For example, if both 'bar' and 'newbar' values are
+set for a key called 'foo', the 'newbar' value would take precedence:
+
+    $ helm upgrade --set foo=bar --set foo=newbar redis ./redis
+`
+
+func newUpgradeCmd(cfg *action.Configuration, logger log.Logger) *cobra.Command {
+	client := action.NewUpgrade(cfg)
+	valueOpts := &values.Options{}
+	var outfmt output.Format
+	var createNamespace bool
+
+	cmd := &cobra.Command{
+		Use:   "upgrade [RELEASE] [CHART]",
+		Short: "upgrade a release",
+		Long:  upgradeDesc,
+		Args:  require.ExactArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client.Namespace = settings.Namespace()
+			helmSettings := helmCli.New()
```

Comment:
Please do not create an instance of the helmCli manually, this includes code that accesses helms environment variables and can cause severe problems if a user runs both `hypper` and `helm` on the same machine.
We have introduced this as composition in https://github.com/rancher-sandbox/hypper/blob/handle_repos_composition/pkg/cli/environment.go#L40 where we try to make sure to overwrite those, please use this as soon as PR #58 is merged.

Reply:
I was currently looking at that part of your PR, will do as soon as its merged!

Reply:
Thank you :)